en/services/timesync_v1.md

```diff
@@ -0,0 +1,45 @@
+# Time Synchronization Protocol v1
+
+This protocol is used to estimate the time offsets between system clocks on two MAVLink systems, in a two-system MAVLink network.
+This is useful because it allows ....
+
+The protocol uses just one message [TIMESYNC](#TIMESYNC), which has two `int64_t` fields, `tc1` and `ts1`. 
+The message is sent out with a timestamp in `ts1` (current unix timestamp *1000).
+This is mirrored back by the remote system, along with its own timestamp.
+The first system can then determine the round-trip time, and estimate the timestamp offset.
+This sequence is run multiple times to remove the transient channel and processor usage from the offset calculation.
+
+The protocol only works when there are only two systems on the MAVLink network.
+This is because [TIMESYNC](#TIMESYNC) is broadcast (cannot be addressed to a single node), so multiple components on the network may respond.
+A recipient has no way to reliably determine if the timestamp in `ts1` is the one it sent out, or is a reply to another system.
```

Comment:
ArduPilot encodes its own system ID in the bottom-most bits of the nano-second field.  I suggest that's a reasonable pattern other GCSs could adopt.

```
    // we add in our own system id try to ensure we only consider
    // responses to our own timesync request messages
    return AP_HAL::micros64()*1000LL + mavlink_system.sysid;
```

Reply:
@peterbarker Thanks very much.  I'd have preferred we did it the "mavlink way" and had system/component ids in the message. Mostly because then you wouldn't be requiring every system on the network to respond, even if you aren't very interested.

Other than the unnecessary traffic on a very busy network this is great because the timestamp is mirrored back and used by the same system - no compatibility issues. @julianoes Are you OK with this?

Reply:
Interesting. PX4 also uses nanoseconds in the time field, so I think we could adopt it.

So that's essentially the same as the `target_system` and `target_component` fields, or the source `sysid`, `compid`?

If it's the same as `target_system` and `target_component` I would strongly suggest that we add that in as an extension.

Or would that create problems that I'm missing?

Reply:
@julianoes ArduPilot just copies in the system_id (not the component id, so you can't sync to a component). The value copied in is the sender id. At the end the value is sent back by the recipient unchanged - so when the original sender gets it back it can tell it is the timestamp it sent out.

So 
1. not the target_system/component being sent. 
2. not quite as flexible as if we did that - since it is only system to system.
3. But less disruptive. Because values are just mirrored back you don't need to think about compatibility with legacy systems or anything really.

Reply:
So what if you have a different component ID?

I think we should tolerate the existing implementation and note it down but extend the message in a way to properly support system and component IDs.

Reply:
@julianoes Thanks. I have proposed the text in https://github.com/mavlink/mavlink-devguide/pull/433/files#r936287659 to reflect the situation now in "timesync v1". This includes the system id.

If we were to add system/component ID then we'd create a v2 version in which you would use the fields as proposed.

I can't add that unless someone is willing to implement it - no stakeholder.

Why should we do it? 
- It is more mavlink, and therefore less of a hack and confusing for all concerned.
- This would work for components too. Is it likely you'll want to have multiple components syncing in one system?
- Reduced bandwidth. You could sync to just one component (say) that you are interested in, and avoid responses from every other system on the network.

I lean towards doing it as "the right thing", but against because I don't think ArduPilot will update.  Unless we see the enough value for the component use case and reduced bandwith, I'm against complicating our different implementations further.

Reply:
> If we were to add system/component ID then we'd create a v2 version in which you would use the fields as proposed

Why not extensions?

Reply:
Sorry, I mean a "v2 version of the protocol". I'd be happy enough with the extensions for the messages.

What I'm worried about here is microservices management. e.g. how a system knows what protocols are supported, what if the extensions are implemented on a sender system but not a reciever etc etc. To me the clearest way to handle that is to call this a new protocol. 

Perhaps I am over thinking it - but it's easy to add two fields as an extension and miss all the other "questions". And as a reader, it is much easier to deprecate the ardupilot way of doing things in a separate doc than mix it in just one doc.

Reply:
Right. The problem I see is that new messages are very slow to get picked up and put the burden on the implementer that now need to support both ways.